frontend/src/concepts/k8s/pods/__tests__/usePodContainers.spec.ts

```diff
@@ -0,0 +1,34 @@
+import { k8sGetResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { mockPipelinePodK8sResource } from '~/__mocks__/mockPipelinePodK8sResource';
+import { PodKind } from '~/k8sTypes';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import usePodContainers from '~/concepts/k8s/pods/usePodContainers';
+
+jest.mock('@openshift/dynamic-plugin-sdk-utils', () => ({
+  k8sGetResource: jest.fn(),
+}));
+
+const k8sGetResourceMock = jest.mocked(k8sGetResource<PodKind>);
+
+describe('usePodContainers', () => {
+  it('should return and fetch pod containers', async () => {
+    k8sGetResourceMock.mockResolvedValue(mockPipelinePodK8sResource({}));
+    const renderResult = testHook(usePodContainers)('test-project', 'test-pod');
+    expect(renderResult).hookToStrictEqual([]);
+    expect(k8sGetResourceMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).hookToHaveUpdateCount(1);
+
+    await renderResult.waitForNextUpdate();
+    expect(k8sGetResourceMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).hookToStrictEqual(mockPipelinePodK8sResource({}).spec.containers);
```

Comment:
Create the mock instance once in the test case and reuse the instance everywhere.